javaharness/java/arcs/android/api/IArcsService.aidl

```diff
@@ -15,9 +16,9 @@ interface IArcsService {
       String arcId,
       String pecId,
       String recipe,
-      String particleId,
-      String particleName,
-      String providedSlotId,
+      in List<String> particleIds,
+      in List<String> particleNames,
+      in List<String> providedSlotIds,
```

Comment:
I don't really love having parallel lists here. This would be better as a `List<ParticleData>` or something like that, but it seems a bit cumbersome to do. I found https://developer.android.com/guide/components/aidl#PassingObjects which suggests you could make ParticleData implement Parcelable. But the ParticleData.particle member will cause problems there...

Reply:
yeah, this is where i started, and I think ideally the entire ArcData would be Parcelable (I think they can be nested, so ParticleData would be Parcelable too).
But then i talked to Ray @cromwellian and he proposed I used AutoValue. But I also have logic in this class. 
And if/when we switch to Kotlin, according to Ray we get the parcelable for free, but then probably need a separate class with the logic vs the one in the API, not sure..

Bottom line, it's not completely clear to me what is the right approach (and also this API will be deprecated once we migrate to use the storage API), and this isn't the highest priority task, so I don't have too much time to spend on it, but OTOH i want to get it done.

So I was hoping to get away with this PR as an incremental progress and a TODO.

Reply:
Fair enough. I guess I'm just generally concerned about the number of hacks and TODOs we've been adding, without actually addressing the underlying code health issues.

What you have here would be ok for now, but I think we should settle on a proper solution like AutoValue/Kotlin/protos sooner rather than later..

Reply:
i agree. 
but i would also like to mention that we did do some refactoring and resolved some of the technical debt after the prototype was done. this is not to say everything is awesome. this is to say we aren't completely ignoring it...